QemuPkg/FrontPageButtons/FrontPageButtons.c

```diff
@@ -0,0 +1,253 @@
+/** @file FrontpageButtons.c
+
+ This module installs the MsButtonServicesProtocol and reports the requested state of Vol+ and
+ Vol- as indicated by the smbios table type3 record version string.
+
+ If the Type3.version string is:
+     "Vol+"  -- Return Vol+ pressed.
+     "Vol-"  -- Return Vol- pressed.
+     any other string, or no string, return neither pressed
+
+  Use only for Qemu platforms which can change smbios every time it boots.
+
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include <Uefi.h>
+
+#include <Protocol/ButtonServices.h>
+#include <Protocol/Smbios.h>
+
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+
+typedef enum {
+  NoButtons     = 0,
+  VolUpButton   = 1,
+  VolDownButton = 2
+} BUTTON_STATE;
+
+#define SMBIOS_VOLUME_UP    "Vol+"
+#define SMBIOS_VOLUME_DOWN  "Vol-"
+
+BUTTON_STATE  gButtonState = NoButtons;

```

Comment:
```suggestion
BUTTON_STATE  mButtonState = NoButtons;

```

Reply:
Refactored the code to not need a global variable.